Parse/src/main/java/com/parse/ParseHttpRequest.java

```diff
@@ -14,8 +14,56 @@
 import java.util.Map;
 
 /** package */ class ParseHttpRequest {
+
+  public enum Method {
+    GET, POST, PUT, DELETE;
+
+    public static Method fromString(String string) {
+      Method method = null;
+      switch (string) {
+        case "GET":
+          method = GET;
+          break;
+        case "POST":
+          method = POST;
+          break;
+        case "PUT":
+          method = PUT;
+          break;
+        case "DELETE":
+          method = DELETE;
+          break;
+        default:
+          break;
+      }
+      return method;
+    }
+
+    @Override
+    public String toString() {
+      String string = null;
+      switch (this) {
+        case GET:
+          string = "GET";
+          break;
+        case POST:
+          string = "POST";
+          break;
+        case PUT:
+          string = "PUT";
+          break;
+        case DELETE:
+          string = "DELETE";
+          break;
+        default:
```

Comment:
Here i'd assume we'd either call `super.toString()` or throw `IllegalStateException` since it cannot happen.